initrd/utils.go

```diff
@@ -55,3 +61,147 @@ func compressFiles(output string, writer *cpio.Writer, reader *os.File) error {
+			if err != nil {
+				return err
+			}
+
+			// Construct the target file path
+			targetPath := filepath.Join(targetDir, header.Name)
+
+			switch header.Typeflag {
+			case tar.TypeDir:
+				// Create directory
+				if err := os.MkdirAll(targetPath, os.FileMode(header.Mode)); err != nil {
+					return err
+				}
+			case tar.TypeReg:
+				// Create parent directory if necessary (optional)
+				if err := os.MkdirAll(filepath.Dir(targetPath), 0o755); err != nil {
+					return err
+				}
+
+				// Create and write the file
+				outFile, err := os.OpenFile(targetPath, os.O_CREATE|os.O_WRONLY, os.FileMode(header.Mode))
+				if err != nil {
+					return err
+				}
+				if _, err := io.Copy(outFile, tarReader); err != nil {
+					outFile.Close()
+					return err
+				}
+				outFile.Close()
+			case tar.TypeSymlink:
+				// Create a symlink
+				if err := os.Symlink(header.Linkname, targetPath); err != nil {
+					return fmt.Errorf("could not create symlink from %s to %s: %w", targetPath, header.Linkname, err)
+				}
+			case tar.TypeLink:
+				// Link needs to point to exact file
+				linkPath := filepath.Join(targetDir, header.Linkname)
+
+				// Create a hard link
+				if err := os.Link(linkPath, targetPath); err != nil {
+					return fmt.Errorf("could not create hard link from %s to %s: %w", targetPath, linkPath, err)
+				}
+			default:
+				log.Printf("Skipping unsupported file type: %c in %s", header.Typeflag, header.Name)
+			}
+		}
+
+		source = targetDir
+	}
+
+	// Open writer for the output file
+	writer, err := os.OpenFile(output, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0o644)
+	if err != nil {
+		return fmt.Errorf("could not open output file: %w", err)
+	}
+
+	err = erofs.Create(writer, source, allRoot)
+	if err != nil {
+		return fmt.Errorf("could not create EROFS filesystem: %w", err)
+	}
+	writer.Close()
+
+	return nil
+}
+
+func isCpioFile(initrd string) (bool, error) {
+	fi, err := os.Open(initrd)
+	if err != nil {
+		return false, fmt.Errorf("could not open file: %w", err)
+	}
+	defer fi.Close()
+
+	reader := cpio.NewReader(fi)
+
+	_, _, err = reader.Next()
+	if err != nil {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func isErofsFile(initrd string) (bool, error) {
+	fi, err := os.Open(initrd)
+	if err != nil {
+		return false, fmt.Errorf("could not open file: %w", err)
+	}
+	defer fi.Close()
+
+	_, err = erofs.Open(fi)
+	if err != nil {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func isTarGzFile(initrd string) (isTar bool, isGz bool, err error) {
+	file, err := os.Open(initrd)
+	if err != nil {
+		return false, false, fmt.Errorf("could not open file: %w", err)
+	}
+	defer file.Close()
+
+	gzReader, err := gzip.NewReader(file)
+	if err == nil {
+		defer gzReader.Close()
+		isGz = true
+	}
+
+	var tarReader *tar.Reader
+	if isGz {
+		tarReader = tar.NewReader(gzReader)
+	} else {
+		tarReader = tar.NewReader(file)
+	}
+	_, err = tarReader.Next()
+	if err == nil || err == tar.ErrHeader {
+		isTar = true
+	}
+	return
+}
```

Comment:
Redundant method `isTarGzFile`, this already exists here:

https://github.com/unikraft/kraftkit/blob/staging/archive/unarchive.go#L46

Reply:
Added this by mistake